cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1980,4 +1982,45 @@ private void deleteProcessMetricsForFlow(String application, String flow) throws
       client.close();
     }
   }
+
+  private BasicArguments getSystemArguments(String namespaceId) {
+    // Get Configs from Cconf
+    Map<String, String> systemConfigsFromCDAP = getSystemArgumentsFromCConf();
+    // Override the Configs from configs at namespace level.
+    return new BasicArguments(getNamespaceResolvedSystemArguments(namespaceId, systemConfigsFromCDAP));
+  }
+
+  private Map<String, String> getSystemArgumentsFromCConf() {
+
+    Map<String, String> configs = Maps.newHashMap();
+
+    // The only config currently as system arguments is Scheduler queue.
+    String schedulerQueue = configuration.get(Constants.AppFabric.APP_SCHEDULER_QUEUE);
+    if (schedulerQueue != null) {
+      configs.put(Constants.AppFabric.APP_SCHEDULER_QUEUE, schedulerQueue);
+    }
+
+    return configs;
+  }
+
+  private Map<String, String> getNamespaceResolvedSystemArguments(String namespaceId, Map<String, String> configs) {
+    Map<String, String> resolvedConfigs = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : configs.entrySet()) {
```

Comment:
The for loop can be replaced by `Map<String, String> resolvedConfigs = Maps.newHashMap(configs)`